### PR TITLE
test(gateway): support array references and empty reponse

### PR DESCRIPTION
**Pull request by PacoDu**
An empty `_entities` response for `LIST_TYPE` currently produces the following error:  `Cannot read property 'json' of undefined`.

### CODE REVIEW

File: lib/gateway.js [hunk 1]
```diff
@@ -146,6 +144,25 @@ function defineResolvers (schema, typeToServiceMap, serviceMap, typeFieldsToServ
   }
 }
 
+function makeEntityTransformData (field) {
+  const isNonNull = field.astNode.type.kind === Kind.NON_NULL_TYPE
+  const leafKind = isNonNull ? field.astNode.type.type.kind : field.astNode.type.kind
```
Comment (PacoDu) [inline lib/gateway.js:149]: There may be a better/more generic way to find theses values but for now it solves the issue nullable/not nullable lists

File: lib/gateway.js [hunk 2]
```diff
@@ -146,6 +144,25 @@ function defineResolvers (schema, typeToServiceMap, serviceMap, typeFieldsToServ
...
+        return isNonNull ? [] : null
+      }
+
+      return response.json.data._entities
+    }
+  }
+
+  return function transformData (response) {
+    return response.json.data._entities[0]
+  }
```
Comment (mcollina) [inline lib/gateway.js:163]: This function can be a top level one, as there is no need to allocate a closure.

File: lib/gateway.js [hunk 3]
```diff
@@ -146,6 +144,25 @@ function defineResolvers (schema, typeToServiceMap, serviceMap, typeFieldsToServ
...
+  const leafKind = isNonNull ? field.astNode.type.type.kind : field.astNode.type.kind
+
+  if (leafKind === Kind.LIST_TYPE) {
+    return function transformData (response) {
+      if (!response) {
+        return isNonNull ? [] : null
+      }
+
+      return response.json.data._entities
+    }
```
Comment (mcollina) [inline lib/gateway.js:158]: I would prefer if we did not create a closure over the ast node, can you move this to two top level functions?

**Comment by PacoDu** [comment]
related to #310

**Comment by PacoDu** [comment]
Strange, idk why the CI failed here, it's working locally

**Review by mcollina — APPROVED** [review]
lgtm

File: test/gateway/schema.js [hunk 4]
```diff
@@ -865,3 +865,281 @@ test('Should support multiple `extends` of the same type in the service SDL', as
...
+        },
+        {
+          name: 'user',
+          url: `http://localhost:${userServicePort}/graphql`
+        }
+      ]
+    }
+  })
+
+  await gateway.listen(0)
```
Comment (mcollina) [inline test/gateway/schema.js:1103]: You can remove this, there is no need as you are using inject()

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**